reconcile/external_resources/aws.py

```diff
@@ -142,9 +175,53 @@ def resolve(self, spec: ExternalResourceSpec) -> dict[str, Any]:
                 spec.provisioner_name, kms_key_id
             ).identifier
 
+        # If not timeouts are set, set default timeouts according to the module reconcile timeout configuration
+        # 5 minutes are substracted to let terraform finish gracefully before the Job is killed.
+        if "timeouts" not in data:
+            data["timeouts"] = {
+                "create": f"{module_conf.reconcile_timeout_minutes - 5}m",
+                "update": f"{module_conf.reconcile_timeout_minutes - 5}m",
+                "delete": f"{module_conf.reconcile_timeout_minutes - 5}m",
+            }
         return data
 
-    def validate(self, resource: ExternalResource) -> None: ...
+    def _get_timeout_value(
+        self,
+        timeout: str,
+    ) -> int:
+        if not re.match(AWSRdsFactory.TIMEOUT_RE, timeout):
+            raise ValueError(
+                f"Invalid RDS instance timeout format: {timeout}. Specify timeout in minutes(m) or hours(h)."
+            )
+        value, unit = timeout[:-1], timeout[-1]
+        return int(value) * AWSRdsFactory.TIMEOUT_UNIT_CONVERSION[unit]
```

Comment:
how about we only support `m` to simplify checks, we don't support full golang time duration format anyway. defaults are documented as `m` too https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance#timeouts

and since the return is minutes, better to rename it to `_get_timeout_minutes`

Reply:
Done.